plutarch-ledger-api/src/Plutarch/LedgerApi/Interval.hs

```diff
@@ -176,16 +191,24 @@ newtype PUpperBound (a :: S -> Type) (s :: S)
       PIsData
     , -- | @since 2.0.0
       PDataFields
-    , -- | @since 2.0.0
-      PEq
-    , -- | @since 2.0.0
-      PPartialOrd
     , -- | @since 2.0.0
       POrd
     , -- | @since 2.0.0
       PShow
     )
 
+-- | @since WIP
+instance (PIsData a, PEnumerable a) => PEq (PUpperBound a) where
+  {-# INLINEABLE (#==) #-}
+  ub1 #== ub2 = (pinclusiveUpperBound # ub1) #== (pinclusiveUpperBound # ub2)
```

Comment:
Do we want this as default comparison? instead of Eq checking if two bounds are strictly equal (by definition)?

Reply:
This is identical to how they're checked by Plutus. Doing it the way you suggest would make it inconsistent with the order semantics.

Reply:
> This is identical to how they're checked by Plutus
Okay. Let's stick to this